pkg/cloudproduct/gke/gke.go

```diff
@@ -84,3 +90,74 @@ func (*gkeAutopilot) SyncPodPortsToGameServer(gs *agonesv1.GameServer, pod *core
 	}
 	return nil
 }
+
+func (*gkeAutopilot) NewPortAllocator(minPort, maxPort int32,
+	_ informers.SharedInformerFactory,
+	_ externalversions.SharedInformerFactory) portallocator.Interface {
+	return &autopilotPortAllocator{minPort: minPort, maxPort: maxPort}
+}
+
+func (*gkeAutopilot) ValidateGameServer(gs *agonesv1.GameServer) []metav1.StatusCause {
+	var causes []metav1.StatusCause
+	for _, p := range gs.Spec.Ports {
+		if p.PortPolicy != agonesv1.Dynamic {
+			causes = append(causes, metav1.StatusCause{
+				Type:    metav1.CauseTypeFieldValueInvalid,
+				Field:   fmt.Sprintf("%s.portPolicy", p.Name),
+				Message: errPortPolicyMustBeDynamic,
+			})
+		}
+	}
+	return causes
+}
+
+type autopilotPortAllocator struct {
+	minPort int32
+	maxPort int32
+}
+
+func (*autopilotPortAllocator) Run(_ context.Context) error        { return nil }
+func (*autopilotPortAllocator) DeAllocate(gs *agonesv1.GameServer) {}
+
+func (apa *autopilotPortAllocator) Allocate(gs *agonesv1.GameServer) *agonesv1.GameServer {
+	if len(gs.Spec.Ports) == 0 {
+		return gs // Nothing to do.
+	}
+
+	var ports []agonesv1.GameServerPort
+	for i, p := range gs.Spec.Ports {
+		if p.PortPolicy != agonesv1.Dynamic {
+			logger.WithField("gs", gs.Name).WithField("PortPolicy", p.PortPolicy).Error(
```

Comment:
Super tiny nit:

```suggestion
			logger.WithField("gs", gs.Name).WithField("portPolicy", p.PortPolicy).Error(
```

(Which made me go back and look at what we do everywhere else, and all fields we have start with a lower case)

Reply:
Heh, I had it as `portPolicy` initially and changed it back, oops. :) Fixed.